### PR TITLE
Simming ai player redux

### PR DESCRIPTION
**Pull request by domino14**

### CODE REVIEW

**Review by martindemello — COMMENTED** [review]
this doesn't look right - it's calling itself

File: ai/bot/bot_player.go [hunk 1]
```diff
@@ -70,25 +70,29 @@ func addBotFields(p *turnplayer.BaseTurnPlayer, conf *config.Config, botType pb.
...
 	return p.TopPlays(plays, numPlays)
 }
 
+func (p *BotTurnPlayer) BestPlay(ctx context.Context) (*move.Move, error) {
+	return p.GenerateMoves(1)[0], nil
+}
+
 func (p *BotTurnPlayer) SetEquityCalculators(calcs []equity.EquityCalculator) {
-	p.calculators = calcs
+	p.SetEquityCalculators(calcs)
```
Comment (martindemello) [inline ai/bot/bot_player.go:95]: meant to leave the comment about calling itself here (stupid github interface!)

Comment (domino14) [reply ai/bot/bot_player.go:95]: oops. you're right. I must not actually be calling that function.

File: montecarlo/montecarlo.go [hunk 2]
```diff
@@ -566,19 +567,23 @@ func (s *Simmer) printStats() string {
...
+		}
 
-		stats += fmt.Sprintf("%-20s%-9d%-16s%-16s\n", play.play.ShortDescription(),
-			play.play.Score(), wpStats, eqStats)
+		statsStr.WriteString(fmt.Sprintf("%-20s%-9d%-16s%-16s%s\n", play.play.ShortDescription(),
+			play.play.Score(), wpStats, eqStats, ignore))
 	}
-	stats += fmt.Sprintf("Iterations: %d (intervals are 95%% confidence)\n", s.iterationCount)
-	return stats
+	statsStr.WriteString(fmt.Sprintf("Iterations: %d (intervals are 99%% confidence, ❌ marks plays cut off early)\n", s.iterationCount))
```
Comment (andy-k) [inline montecarlo/montecarlo.go:585]:
```go
package main

import (
	"fmt"
	"strings"
)

func main() {
	//ss := new(strings.Builder) // unnecessary alloc
	var ss strings.Builder // avoid new()

	ss.WriteString(fmt.Sprintf("Hello %d", 123)) // unnecessary alloc/copy
	fmt.Fprintf(&ss, " this is %d", 456)         // can directly Fprintf

	fmt.Println(ss.String())
}
```

**Pull request merged**